src/lib/auth/README.md

```diff
@@ -0,0 +1,44 @@
+## Authentication Library
+Initialize authentication library (authenticator). A strategy will be prepared based on the authentication type.
+
+**The library is promised based, all exposed methods return promises.**
+
+The module acts like a singleton. Both the strategy and the authentication options are saved in
+the package's object which allows us to only initialize the library once. Meaning that after we
+initialize the library it will be shared throughout the entire codebase without the need of
+re-initialization.
+
+Currently implemented strategies are:
+ - `wpcom-oauth` - regular wpcom oAuth process using `wpcom-oauth-cors` package
+ - `wpcom-oauth-by-token` - regular wpcom oAuth based on given token using `wpcom-xhr-request`
+ - `wpcom-proxy-iframe` - proxy iframe wpcom authentication using Calypso's wpcom object
+
+ ## API
+ `init( settings )` - Initialize authentication library (authenticator). A strategy will be prepared based on the authentication type
+
+ #### Available settings
+
+| Key | Value | Required | Default | Available Values | Description |
+| --- | --- | --- | --- | --- | --- |
+| `type` | string | Required | - | `wpcom-oauth` \| `wpcom-oauth-by-token` \| `wpcom-proxy-iframe` | A valid implemented strategy type |
+| `options` | object | Optional | - | - | Set of authentication options
+| `options.token` | string | Required by `wpcom-oauth-by-token` | - | - | oauth accessToken
+| `options.proxy` | object | Required by `wpcom-proxy-iframe` | - | - | wpcom auth proxy object
```

Comment:
I think I see now how this works! I guess we need to expose the `oauth_client_id` through the `options` object, not through the config, as per the reasons I mentioned above.

Reply:
I hesitate between exposing the `oauth_client_id` and dropping the `wpcom-oauth` strategy altogether as per [this comment](https://github.com/Automattic/happychat-client/pull/188/files#r178528892). I lean towards not having that strategy at the moment. We are not using it in any current or planned host and makes things more confusing. We can always expose it later if/when that's needed.

Reply:
It was used only in the standalone example, will remove it.